rskj-core/src/test/java/co/rsk/peg/BridgeSupportIT.java

```diff
@@ -192,17 +199,114 @@ void testInitialChainHeadWithoutBtcCheckpoints() throws Exception {
     void testInitialChainHeadWithBtcCheckpoints() throws Exception {
         BridgeConstants bridgeTestNetConstants = BridgeTestNetConstants.getInstance();
 
+        BridgeSupport bridgeSupport = arrangeBridgeSupport(bridgeTestNetConstants,
+            activationsBeforeForks);
+
+        // Force instantiation of blockstore
+        bridgeSupport.getBtcBlockchainBestChainHeight();
+
+        InputStream checkpointsStream = bridgeSupport.getCheckPoints();
+        CheckpointManager manager = new CheckpointManager(bridgeTestNetConstants.getBtcParams(), checkpointsStream);
+        long time = bridgeSupport.getActiveFederation().getCreationTime().toEpochMilli() - 604800L; // The magic number is a substraction CheckpointManager does when getting the checkpoints.
+        StoredBlock checkpoint = manager.getCheckpointBefore(time);
+
+        assertEquals(checkpoint.getHeight(), bridgeSupport.getBtcBlockchainBestChainHeight());
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        {
+            "12-byte-chainwork.production.checkpoints",
+            "12-byte-chainwork-mix-format.production.checkpoints"
+        }
+    )
+    void testInitialChainHeadWithBtcCheckpoints_whenCheckpointsWith12BytesChainWork_before_RSKIP454_ok(
```

Comment:
can we add the arrange, act, assert sections as we've been doing? I find the tests a bit hard to follow

Reply:
Done